### PR TITLE
Re-implement file logger

### PR DESCRIPTION
**Pull request by joseferben**

### CODE REVIEW

**Comment by joseferben** [comment]
I re-added the file logger here, in case log aggregation will be done this way.

File: pool/app/logger/logger.ml [hunk 1]
```diff
@@ -118,9 +118,99 @@ let format_reporter
   { Logs.report }
 ;;
 
+let get_log_level () =
+  match Sys.getenv_opt "LOG_LEVEL" with
```
Comment (aronerben) [inline pool/app/logger/logger.ml:122]: Could also use `Sihl.Configuration.read_string` here?

File: pool/app/logger/logger.ml [hunk 2]
```diff
@@ -118,9 +118,99 @@ let format_reporter
...
+  in
+  let app, app_flush = buf () in
+  let err, err_flush = buf () in
+  let report src level ~over k msgf =
+    let k _ = k () in
+    let write () =
+      let name =
+        match[@warning "-4"] level with
+        | Logs.Error -> logs_dir ^ "/error.log"
+        | _ -> logs_dir ^ "/app.log"
```
Comment (aronerben) [inline pool/app/logger/logger.ml:156]: Please be exhaustive here (`| App | Warning | Info | Debug -> ...`) and get rid of suppression of warning 4.

File: pool/app/logger/logger.ml [hunk 3]
```diff
@@ -118,9 +118,99 @@ let format_reporter
...
+        Lwt_io.open_file
+          ~flags:[ Unix.O_WRONLY; Unix.O_CREAT; Unix.O_APPEND ]
+          ~perm:0o777
+          ~mode:Lwt_io.Output
+          name
+      in
+      let%lwt () =
+        match[@warning "-4"] level with
+        | Logs.Error -> Lwt_io.write log (err_flush ())
+        | _ -> Lwt_io.write log (app_flush ())
```
Comment (aronerben) [inline pool/app/logger/logger.ml:168]: Same exhaustiveness here please

File: pool/app/logger/logger.ml [hunk 4]
```diff
@@ -118,9 +118,99 @@ let format_reporter
...
+      in
+      Lwt_io.close log
+    in
+    let unblock () =
+      over ();
+      Lwt.return_unit
+    in
+    Lwt.finalize write unblock |> Lwt.ignore_result;
```
Comment (aronerben) [inline pool/app/logger/logger.ml:176]: Use the `lwt.finally` ppx here please to ensure stack traces (https://github.com/uzh/pool/commit/7867e2c68906ced3199a66d388caa2bd48f85e0d#diff-1f214c71db6e40598f3df14bec1437e3e41f5359bcd83bfdcab6cfc527ebde48L114-R124 see here for example)

File: pool/app/logger/logger.ml [hunk 5]
```diff
@@ -118,9 +118,99 @@ let format_reporter
...
+    msgf
+    @@ fun ?header ?tags fmt ->
+    let ppf =
+      match[@warning "-4"] level with
+      | Logs.Error -> Format.formatter_of_buffer err
+      | _ -> Format.formatter_of_buffer app
```
Comment (aronerben) [inline pool/app/logger/logger.ml:182]: Same exhaustiveness here please

**Review by aronerben — CHANGES_REQUESTED** [review]

File: pool/app/logger/logger.ml (hunk 1 above)
Comment (joseferben) [reply pool/app/logger/logger.ml]: Done

File: pool/app/logger/logger.ml (hunk 2 above)
Comment (joseferben) [reply pool/app/logger/logger.ml]: Done

File: pool/app/logger/logger.ml (hunk 3 above)
Comment (joseferben) [reply pool/app/logger/logger.ml]: Done

File: pool/app/logger/logger.ml (hunk 4 above)
Comment (joseferben) [reply pool/app/logger/logger.ml]: Done

**Comment by joseferben** [comment]
@aronerben All done

**Comment by joseferben** [comment]
Please merge if ok

**Pull request merged**